activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -456,10 +454,16 @@ def find_some(ids)
       # 11 ids with limit 3, offset 9 should give 2 results.
       if offset_value && (ids.size - offset_value < expected_size)
         expected_size = ids.size - offset_value
+      else
+        ids = ids.first(expected_size) unless self.values[:order]
```

Comment:
I think `order_values.empty?` is better here. I saw it in other places in code and it is the fastest method.